background.ts

```diff
@@ -454,6 +454,15 @@ Electron.ipcMain.on('api-get-credentials', () => {
   mainEvents.emit('api-get-credentials');
 });
 
+Electron.ipcMain.handle('api-get-credentials', async() => {
+  const credentials = await (new Promise<void>((resolve) => {
+    mainEvents.on('api-credentials', resolve);
```

Comment:
This doesn't sound right -- we don't do this anywhere else in the code, where we create an event-handler inside another event-handler. At first I thought you want to handle this event globally, but then what would you do. Here it looks like you're just waiting for a second event to arrive after the `api-get-credentials` event arrives. This either is right and needs to be documented, or something else needs to happen here. Or I need to have the current setup explained.

Reply:
This pattern exists elsewhere in several places in `src/window/index.ts` and once more in `background.ts`. 

There needs to be an an invokable way to get our credentials so that await the response in the renderer before attempting to communicate with the preferences api. The only way to get credentials out of `src/main/commandServer/httpCommandServer.ts` is events, so resolving on `api-credentials` gets us what we need. Maybe it would make more sense to refactor `src/main/commandServer/httpCommandServer.ts`.